Decred Wallet/Features/History/TransactionHistoryViewController.swift

```diff
@@ -55,140 +60,135 @@ class TransactionHistoryViewController: UIViewController {
+            break
+        case DcrlibwalletTxDirectionReceived:
+            self.filteredTransactions = self.allTransactions.filter{$0.Direction == DcrlibwalletTxDirectionReceived && $0.Type == DcrlibwalletTxTypeRegular}
+            self.btnFilter.setTitle(LocalizedStrings.received.appending("(\(self.filteredTransactions.count))"), for: .normal)
+            break
+        case DcrlibwalletTxDirectionTransferred:
+            self.filteredTransactions = self.allTransactions.filter{$0.Direction == DcrlibwalletTxDirectionTransferred && $0.Type == DcrlibwalletTxTypeRegular}
+            self.btnFilter.setTitle(LocalizedStrings.yourself.appending("(\(self.filteredTransactions.count))"), for: .normal)
+            break
+        case DcrlibwalletTxFilterStaking:
+            self.filteredTransactions = self.allTransactions.filter{$0.Type == DcrlibwalletTxTypeRevocation || $0.Type == DcrlibwalletTxTypeTicketPurchase || $0.Type == DcrlibwalletTxTypeVote }
+            self.btnFilter.setTitle(LocalizedStrings.staking.appending("(\(self.filteredTransactions.count))"), for: .normal)
+            break
+        case DcrlibwalletTxFilterCoinBase:
+            self.filteredTransactions = self.allTransactions.filter{$0.Type == DcrlibwalletTxTypeCoinBase}
+            self.btnFilter.setTitle("COINBASE (\(self.filteredTransactions.count))", for: .normal)
+            break
+        default:
+            self.btnFilter.setTitle(LocalizedStrings.all.appending("(\(self.allTransactions.count))"), for: .normal)
+            break
         }
+        self.refreshControl.endRefreshing()
     }
     
     override func viewDidDisappear(_ animated: Bool) {
         super.viewDidDisappear(animated)
-        self.visible = false
         dismiss(animated: true, completion: nil)
     }
     
     @objc func handleRefresh(_ refreshControl: UIRefreshControl) {
-        self.loadTransactions()
-    }
-    
-    func loadTransactions() {
-        self.mainContens.removeAll()
-        self.Filtercontent.removeAll()
-        
-        // use count = 0 to return all transactions
-        AppDelegate.walletLoader.wallet?.transactionHistory(count: 0) { transactions in
-            self.refreshControl.endRefreshing()
-            
-            if transactions == nil || transactions!.count == 0 {
-                self.showNoTransactions()
-                return
-            }
-            
-            self.mainContens = transactions!
-            self.tableView.backgroundView = nil
-            self.tableView.separatorStyle = .singleLine
-            self.reloadTxsForCurrentFilter()
-        }
+        self.reloadTxsForCurrentFilter()
     }
     
     func showNoTransactions() {
         let label = UILabel(frame: CGRect(x: 0, y: 0, width: self.tableView.bounds.size.width, height: self.tableView.bounds.size.height))
         label.text = LocalizedStrings.noTransactions
         label.textAlignment = .center
+        self.refreshControl.endRefreshing()
```

Comment:
Move this to after line 148. Add an empty line between line 148 and this `endRefreshing` line to show that the first set of codes (relating to the no tx label) is not completely related to this `endRefreshing` line.

Reply:
fixed in e594317